backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/CreateSnapshotFromTemplateCommand.java

```diff
@@ -89,6 +92,9 @@ protected void endWithFailure() {
             if (diskImageDynamicDao.get(getDestinationDiskImage().getImageId()) != null) {
                 diskImageDynamicDao.remove(getDestinationDiskImage().getImageId());
             }
+            if (imageDao.get(getDestinationDiskImage().getImageId()) != null) {
```

Comment:
will it answer on some of the comment above? should you check if it succeed before removing as written?

Reply:
OK, although i do think it might be better to insert into the remove image command. but, since we already have other DB removals like that (baseDiskDao and diskImageDynamicDao)...